### PR TITLE
[GH-12955] Migrate 'components/admin_console/highlight.jsx' to TypeScript

### PR DESCRIPTION
**Pull request by larkox**
#### Summary

Migrate 'components/admin_console/highlight.jsx' to TypeScript
#### Ticket Link

https://github.com/mattermost/mattermost-server/issues/12955
#### Related Pull Requests

None

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,8 @@
     "@types/enzyme": "3.10.3",
     "@types/history": "^4.7.3",
     "@types/jest": "24.0.17",
+    "@types/lodash": "^4.14.144",
```
Comment (jespino) [inline package.json]: We normally use fixed versions, please remove the `^` at the beginning.

File: package.json [hunk 2]
```diff
@@ -9,6 +9,8 @@
...
+    "@types/mark.js": "^8.11.4",
```
Comment (jespino) [inline package.json]: same here

File: components/admin_console/highlight.tsx [hunk 3]
```diff
@@ -6,20 +6,23 @@ import PropTypes from 'prop-types';
...
+export default class Highlight extends React.Component<Props> {
+    private markInstance? : Mark;
+    private ref : React.RefObject<HTMLDivElement>;
 
-    constructor(props) {
+    public constructor(props : Props) {
         super(props);
-        this.markInstance = null;
-        this.ref = React.createRef();
+        this.markInstance = undefined;
```
Comment (jespino) [inline components/admin_console/highlight.tsx]: I think this line is not needed, by default an uninitialized attribute should be undefined.

File: package.json (hunk 1 above)
Comment (larkox) [reply package.json]: Thanks! Noted for future contributions. Fixed on commit 1d2ecb1.

File: package.json (hunk 2 above)
Comment (larkox) [reply package.json]: Thanks! Also fixed on commit 1d2ecb1.

File: components/admin_console/highlight.tsx (hunk 3 above)
Comment (larkox) [reply components/admin_console/highlight.tsx]: Thanks! Fixed on commit 1d2ecb1.

**Comment by jespino** [comment]
Merged! Thanks @larkox! :confetti_ball:

**Pull request merged**